src/oidc_child/oidc_child_json.c

```diff
@@ -0,0 +1,505 @@
+
+    return json_integer_value(tmp);
+}
+
+static char *get_json_scope(TALLOC_CTX *mem_ctx, json_t *root, const char *attr)
+{
+    json_t *tmp;
+    json_t *s;
+    size_t index;
+    char *str = NULL;
+
+    tmp = json_object_get(root, attr);
+    if (!json_is_array(tmp)) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Result does not contain the '%s' array.\n", attr);
+        return NULL;
+    }
+
+    json_array_foreach(tmp, index, s) {
+        if (!json_is_string(s)) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to read supported scopes.\n");
+            talloc_free(str);
+            return NULL;
+        }
+
+        if (str == NULL) {
+            str = talloc_strdup(mem_ctx, json_string_value(s));
+        } else {
+            str = talloc_asprintf_append(str, "%%20%s", json_string_value(s));
+        }
+        if (str == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to copy '%s' string.\n", attr);
+            return NULL;
+        }
+
+    }
+
+    return str;
+}
+
+static errno_t get_endpoints(json_t *inp, struct devicecode_ctx *dc_ctx)
+{
+    int ret;
+
+    dc_ctx->device_authorization_endpoint = get_json_string(dc_ctx, inp,
+                                               "device_authorization_endpoint");
+    if (dc_ctx->device_authorization_endpoint == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing device_authorization_endpoint in "
+                                   "openid configuration.\n");
+        ret = EINVAL;
+        goto done;
+    }
+    dc_ctx->token_endpoint = get_json_string(dc_ctx, inp, "token_endpoint");
+    if (dc_ctx->token_endpoint == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing token_endpoint in openid "
+                                   "configuration.\n");
+        ret = EINVAL;
+        goto done;
+    }
+    dc_ctx->userinfo_endpoint = get_json_string(dc_ctx, inp,
+                                                "userinfo_endpoint");
+    if (dc_ctx->userinfo_endpoint == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing userinfo_endpoint in openid "
+                                   "configuration.\n");
+        ret = EINVAL;
+        goto done;
+    }
+
+    dc_ctx->jwks_uri = get_json_string(dc_ctx, inp, "jwks_uri");
+    if (dc_ctx->jwks_uri == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing jwks_uri in openid "
+                                   "configuration.\n");
+    }
+
+    dc_ctx->scope = get_json_scope(dc_ctx, inp, "scopes_supported");
+    if (dc_ctx->scope == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Missing scopes in openid "
+                                   "configuration.\n");
+    }
+
+    ret = EOK;
+done:
+    return ret;
+}
+
+errno_t get_issuer(struct devicecode_ctx *dc_ctx,
+                   const char *issuer_name)
+{
+    int ret;
+    json_t *inp = NULL;
+    json_t *issuers = NULL;
+    json_error_t json_error;
+    json_t *i;
+    json_t *name;
+    size_t index;
+
+    inp = json_load_file("issuers", 0, &json_error);
```

Comment:
Who populates this file?

Reply:
I used this file for testing to easily switch between different IdP. But it is not needed anymore, so I removed all related code.